debian/snapd.postinst

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+set -e
+systemctl daemon-reload
```

Comment:
should be fairly straightforward (but involved enough to be a separate branch): if in classic, tell the outer systemctl to restart snapd (with `--nowait`). If not in classic, do the reboot dance.

I think all of that is new code tho.